File: tasks/smoketests/bootstrap/extension/extension.js
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

'use strict';

Object.defineProperty(exports, "__esModule", { value: true });
const vscode = require("vscode");
function activate(context) {
    const statusBarItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left, 10000000);
    statusBarItem.command = 'workbench.action.quickOpen';
    statusBarItem.text = 'PySmoke';
    statusBarItem.tooltip = 'PySmoke';
    statusBarItem.show();

    context.subscriptions.push(statusBarItem);

    const ext = vscode.extensions.getExtension('ms-python.python');
    if (!ext.isActive){
        ext.activate();
    }

    vscode.commands.registerCommand('smoketest.activatePython', async ()=>{
        const ext = vscode.extensions.getExtension('ms-python.python');
        if (!ext.isActive){
            await ext.activate();
        }
        vscode.window.showInformationMessage('Python Extension Activated');
    })
}

exports.activate = activate;
function deactivate() {
    // Do nothing.
}
exports.deactivate = deactivate;

File: tasks/smoketests/vscode/extension.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.


from . import documents, notifications, quick_open, status_bar


def load_python_extension(context):
    documents.create_new_untitled_file(context, "Python")
    last_error = None
    for _ in range(5):
        quick_open.select_command("Activate Python Extension")
        try:
            notifications.wait_for_message("Python Extension Activated")
            break
        except Exception as ex:
            last_error = ex
            continue
    else:
        raise SystemError("Failed to activate extension") from last_error
    status_bar.wait_for_python_statusbar(context)
    notifications.clear(context)
